lib/charms/mongodb/v0/mongodb_backups.py

```diff
@@ -198,7 +228,15 @@ def _on_restore_action(self, event) -> None:
             event.fail(f"Cannot restore backup {pbm_status.message}.")
             return
 
-        # TODO restore backup
+        # sometimes when we are trying to restore pmb can be resyncing, so we need to retry
+        try:
+            self._try_to_restore(backup_id)
+            event.set_results({"restore-status": "restore started"})
+            self.charm.unit.status = MaintenanceStatus()
```

Comment:
Should we inform the user why the unit is in `MaintenanceStatus`?
```suggestion
            self.charm.unit.status = MaintenanceStatus("restore started/running")
```

Reply:
Yep, we definitely should